### PR TITLE
Reset client secret

### LINKED ISSUE #1221: Implement My Profile - My Applications reset secret

### LINKED ISSUE #1376: My App Secret
Users need to be able to get their secret after an app is created.  I would propose it being in a modal, only because it re-instills the understanding that the access to it is fleeting, and how they have to reset to get a new secret.

### PR DESCRIPTION
**Pull request by eatonphil**
Depends on #1368, closes #1221, closes #1376 

screen shot 2017-03-14 at 8 37 27 am

### CODE REVIEW

**Comment by na3d** [comment]
This pops up right after My App is created too, right?

**Comment by eatonphil** [comment]
> This pops up right after My App is created too, right?

Yep

**Comment by mparke** [comment]
secret is cut off?

**Comment by eatonphil** [comment]
> secret is cut off?

It scolls

File: src/api/apiActionReducerGenerator.js [hunk 1]
```diff
@@ -283,10 +283,10 @@ function _thunkFetch(method) {
     async (dispatch, getState) => {
       const state = getState();
       const { token } = state.authentication;
-      await fetch(token, url, {
+      return await (await fetch(token, url, {
```
Comment (eatonphil) [inline src/api/apiActionReducerGenerator.js]: The first await is for the fetch overall, the second await is for the json function. They are both promises.

Comment (mparke) [inline src/api/apiActionReducerGenerator.js]: can we break into multiple lines for readability?

File: src/profile/integrations/components/CreatePersonalAccessToken.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import { title } from './OAuthScopes';
+import { tokens } from '~/api';
+import SelectExpiration from '../../components/SelectExpiration';
+
+export async function renderSecret(label, verb, secret) {
+  const { dispatch } = this.props;
+  const close = () => dispatch(hideModal());
+
+  await dispatch(showModal(
+    `${title(label)} ${verb}`,
```
Comment (mparke) [inline src/profile/integrations/components/CreatePersonalAccessToken.js]: `_.capitalize`

title is removed in another branch

File: src/profile/integrations/components/CreatePersonalAccessToken.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    this.renderSecret = renderSecret.bind(this);
+
+    this.state = {
+      ...OAUTH_SCOPES.reduce((object, scope) => ({
+        ...object,
+        [scope]: OAUTH_SUBSCOPES[OAUTH_SUBSCOPES.length - 1],
+      }), {}),
+      errors: {},
+      label: '',
+      expiry: '0',
```
Comment (mparke) [inline src/profile/integrations/components/CreatePersonalAccessToken.js]: string?

**Comment by abemassry** [comment]
The cancel is redundant on the final screen, the client secret has already been reset, so there is nothing to cancel. Otherwise it works well, 👍

**Comment by eatonphil** [comment]
> The cancel is redundant on the final screen

That is a good point... Will fix.

**Pull request merged**